tests/test_adapter.py

```diff
@@ -383,6 +388,24 @@ def test_called_and_called_count(self):
         self.assertTrue(self.adapter.called)
         self.assertFalse(m.called_once)
 
+    def test_called_and_reset(self):
+        m = self.adapter.register_uri('GET', self.url, text='resp')
+        call_count = 3
+
+        for _ in range(0, call_count):
+            self.session.get(self.url)
+
+        # Verify count is expected value for adapter and matcher
+        self.assertEqual(self.adapter.call_count, call_count)
+        self.assertEqual(m.call_count, call_count)
+
+        with patch.object(m, 'reset_mock') as m_matcher_reset:
```

Comment:
similarly, i think just loop through the matchers and assert call_count == 0. we don't need to test the actual code flow.